src/App.tsx

```diff
@@ -44,221 +60,52 @@ export default function App() {
-                });
-                return client;
-            }),
-            (e) => {
-                console.error('Wallet Connect: init error', e);
-                return (e as Error).message;
-            }
-        ).then(setWalletconnect2Client);
-    }, []);
-
-    // Ping Wallet Connect periodically.
-    // TODO Move to WC component.
-    useEffect(() => {
-        if (walletconnect2Client && walletconnect2ConnectedSession) {
-            console.log('setting up ping loop');
-            const interval = setInterval(() => {
-                // console.debug("attempting to ping");
-                walletconnect2Client
-                    .asyncAndThen((c) => {
-                        return ResultAsync.fromPromise(
-                            c.ping({ topic: walletconnect2ConnectedSession.topic }),
-                            (e) => `${e} (${typeof e})`
-                        );
-                    })
-                    .then((r) =>
-                        r.match(
-                            () => {
-                                // console.debug("ping successful");
-                            },
-                            (e) => {
-                                // eslint-disable-next-line no-console
-                                console.error(`ping failed: ${e}`);
-                                setWalletconnect2ConnectionError(e);
-                            }
-                        )
-                    );
-            }, PING_INTERVAL_MS);
-            return () => {
-                // eslint-disable-next-line no-console
-                console.debug('tearing down ping loop');
-                clearInterval(interval);
-            };
-        }
-        return undefined;
-    }, [walletconnect2Client, walletconnect2ConnectedSession]);
-
-    const canUpdate = useMemo(
-        // TODO Give reason?
-        () => {
-            if (wallet === 'browserwallet') {
-                return Boolean(browserwalletConnectedAccount);
-            }
-            if (wallet === 'walletconnect2') {
-                return Boolean(walletconnect2ConnectedSession);
-            }
-            return false;
-        },
-        [wallet, browserwalletConnectedAccount, walletconnect2ConnectedSession]
-    );
-    // TODO Need an interface ('canSmash', 'handleSubmitDeposit', etc.) and a function for mapping from wallet to implementation.
-    const canSmash = useMemo(() => {
-        if (wallet === 'browserwallet' && contract) {
-            return browserwalletConnectedAccount === contract.owner.address;
-        }
-        if (wallet === 'walletconnect2' && walletconnect2ConnectedSession && contract) {
-            return resolveAccount(walletconnect2ConnectedSession) === contract.owner.address;
-        }
-        return false;
-    }, [wallet, browserwalletConnectedAccount, walletconnect2ConnectedSession, contract]);
+    // TODO Need an interface ('canSmash', 'handleSubmitDeposit', etc.).
```

Comment:
```suggestion
```
I guess the interface is the Piggybank Props. (Don't see the point of this TODO :thinking: )

Reply:
Addressed in followup: https://github.com/Concordium/concordium-dapp-piggybank/pull/4